src/openms/source/CHEMISTRY/AASequence.cpp

```diff
@@ -176,11 +178,16 @@ namespace OpenMS
 
   EmpiricalFormula AASequence::getFormula(Residue::ResidueType type, Int charge) const
   {
-    EmpiricalFormula ef;
+
+    // x/c-ion for a single residue is not defined
+    if (peptide_.size() == 1 && (type == Residue::XIon || type == Residue::CIon))
```

Comment:
No strong opinion on this but also prefer to check for >=1. 
Returning the empty EF is not incorrect but might hide some hard to find errors.